packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl

```diff
@@ -25,6 +25,18 @@ apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
```

Comment:
Is it something to warn about if the developer doesn't specify it in local.properties? I suppose adding this warning could help with [Issue #95514](https://github.com/flutter/flutter/issues/95514), though. Is that your reasoning?

Same comment for `flutterTargetSdkVersion`.

Reply:
Exactly that, it could prevent similar issues in the future. Although, since now there is mention of the `local.properties` on the flutter website docs, I don't know if it's that necessary anymore.

Reply:
Oh gotcha! Could we possibly move the warning to [`flutter.gradle`](https://github.com/flutter/flutter/blob/stable/packages/flutter_tools/gradle/flutter.gradle) and only show it when not defining it will be a problem, i.e. when the `minSdkVersion` of a dependency is higher than that of the flutter project?

Reply:
Certainly, I gonna need a little bit of time to understand how `flutter.gradle` works but I am on it. Let me get this straight, ideally, what would be the conditions to trigger the custom warning about `local.properties`?

Reply:
It seems like if the minSdkVersion of a dependency is greater than that of the mindSdkVersion defined in `flutter.gradle`, we show the warning so that the person knows how to fix the error that will appear. You can even add a link to your update on the website to the warning for context!

It should be similar to the way we compare the `compileSdkVersion`s of the project and its plugin dependencies right now (see [here](https://github.com/flutter/flutter/blob/61ac285608cd3c3c003507a572b34fad893a21d4/packages/flutter_tools/gradle/flutter.gradle#L450)), but let me know if I can help!

Reply:
Thank you so much, that should be more than enough for me to get started. I will let you know if I get stuck, thanks!!

Reply:
@camsim99 Updated the PR description, images. I changed `flutter.gradle` and also `gradle_errors.dart`. 

Thank you for reviewing.